src/app/globals.css

```diff
@@ -204,3 +207,36 @@ pre[class*='language-'] {
 
   @apply mr-[calc(var(--pad)/2)];
 }
+
+div[class*='container-language-'] {
+  @apply relative;
+}
+div[class*='container-language-']:hover .copy,
+div[class*='container-language-'] .copy:focus {
+  @apply opacity-100;
+}
+div[class*='container-language-']:hover .copy:not(.copied):hover {
+  @apply bg-surface-variant-dark;
+}
+div[class*='container-language-'] .copy {
+  direction: ltr;
+  background-image: var(--icon-copy);
+  background-size: 20px;
+  transition:
+    border-color 0.25s,
+    background-color 0.25s,
+    opacity 0.25s;
+
+  @apply bg-inverse-surface-light absolute right-4 top-4 z-10 size-10 cursor-pointer rounded border border-white/25 bg-center bg-no-repeat opacity-0;
+}
+div[class*='container-language-'] .copy.copied {
+  background-image: var(--icon-copied);
+
+  @apply rounded-l-none;
+}
+div[class*='container-language-'] .copy.copied:before {
+  transform: translate(calc(-100% - 1px));
+  content: 'Copied';
+
+  @apply bg-inverse-surface-light relative -top-px flex h-10 w-fit items-center justify-center whitespace-nowrap rounded-l border border-r-0 border-white/25 px-2.5 text-center text-xs font-medium text-[#808080];
+}
```

Comment:
can you move this into `src/components/mdx/CopyButton/CopyButton.tsx` ?

Reply:
@abernier sure